core/src/test/java/org/kohsuke/stapler/RequestImplTest.java

```diff
@@ -42,15 +47,55 @@
 
 /**
  * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ * @author Nikolas Falco
  */
 public class RequestImplTest {
 
+    @Test
+    public void verify_JSON_bind_work_with_setter_that_accept_object_type() throws Exception {
+        final Stapler stapler = new Stapler();
+        stapler.setWebApp(new WebApp(Mockito.mock(ServletContext.class)));
+        final MockRequest mockRequest = mockRequest(generateMultipartData());
+        RequestImpl req = new RequestImpl(stapler, mockRequest, Collections.<AncestorImpl>emptyList(), null);
+
+        JSONObject json = new JSONObject();
+        json.put("stapler-class", SetterObject.class.getName());
+        json.put("$class", SetterObject.class.getName());
+        json.put("choices", "1\n2\n3");
+
+        SetterObject o = req.bindJSON(SetterObject.class, json);
```

Comment:
I do not understand what you mean

Reply:
Sorry, just asking whether this was the line of the test which throws an error if you do not have the code fix in place.

Reply:
yes this line will trigger the `TypePair.convertJSON` where there is the fix. The will be triggered two times:
the first one to build the `SetterObject` class and the second one to build the arguments for the `SetterObject.setChoices` method. Without the changes it will throw a `throw new IllegalArgumentException("Unable to convert to "+type);`